File: src/0941.valid-mountain-array.978/solution.ts
function validMountainArray (A: number[]): boolean {
    if (A.length < 3) {
        return false;
    }
    let hasDecreasing = false;
    for (let i = 1; i < A.length; i++) {
        if (A[i] > A[i - 1]) {
            continue;
        }
        if (A[i] === A[i - 1]) {
            return false;
        }
        if (i === 1) {
            return false;
        }
        hasDecreasing = true;
        for (let j = i + 1; j < A.length; j++) {
            if (A[j] >= A[j - 1]) {
                return false;
            }
        }
        break;
    }
    return hasDecreasing;
}

File: src/0941.valid-mountain-array.978/solution.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        if(arr.length<3){
            return false;
        }
        int index= 1 ;
        while (index<arr.length){
            if(arr[index]>arr[index-1]){
                index++;
            }else if(arr[index] == arr[index-1]){
                return false;
            }else{
                break;
            }

        }
        if(index == arr.length || index == 1){
            return false;
        }
        while (index<arr.length){
            if(arr[index]<arr[index-1]){
                index++;
            }else{
                return false;
            }

        }


        return true;
    }
}